src/app/complex/complex-results/complex-navigator/table-structure/table-header/table-header.component.ts

```diff
@@ -0,0 +1,148 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {ComplexSearchResult} from '../../../../shared/model/complex-results/complex-search.model';
+import {TableInteractorColumnComponent} from '../table-interactor-column/table-interactor-column.component';
+import {Element} from '../../../../shared/model/complex-results/element.model';
+
+@Component({
+  selector: 'cp-table-header',
+  templateUrl: './table-header.component.html',
+  styleUrls: ['./table-header.component.css'],
+})
+
+export class TableHeaderComponent implements OnInit {
+  @Input() complexSearch: ComplexSearchResult;
+  @Input() interactorsTable: TableInteractorColumnComponent;
+
+  constructor() {
+  }
+
+  ngOnInit(): void {
+    this.classifyComplexesSize();
+    this.classifyComplexesSimilarities(this.complexSearch);
```

Comment:
What does the second method do? I understand `classifyComplexesSize`, but not `classifyComplexesSimilarities`.

Reply:
ClassifyComplexSimilarities is used to cluster the complexes by number of similarities, it puts the similar complexes next to each other in the list which is used to sort the complexes